### PR TITLE
refactor: Migrate EventService to TypeScript

### PR DESCRIPTION
**Pull request by AndyLnd**

### CODE REVIEW

File: src/script/event/EventService.ts [hunk 1]
```diff
@@ -48,10 +65,13 @@ export class EventService {
    * @param {string[]} eventIds ID of events to retrieve
    * @returns {Promise<Object[]>} Resolves with the stored records
    */
-  async loadEvents(conversationId, eventIds) {
+  async loadEvents(conversationId: string, eventIds: string[]): Promise<DBEvents> {
     if (!conversationId || !eventIds) {
       this.logger.error(`Cannot get events '${eventIds}' in conversation '${conversationId}' without IDs`);
-      throw new ConversationError(BaseError.TYPE.MISSING_PARAMETER, BaseError.MESSAGE.MISSING_PARAMETER);
+      throw new ConversationError(
+        BaseError.TYPE.MISSING_PARAMETER as BASE_ERROR_TYPE,
```
Comment (ffflorian) [inline src/script/event/EventService.ts]:
```suggestion
        BASE_ERROR_TYPE.MISSING_PARAMETER,
```

File: src/script/event/EventService.ts [hunk 2]
```diff
@@ -81,14 +101,16 @@ export class EventService {
...
+   * @param conversationId ID of conversation
+   * @param eventId ID of event to retrieve
    */
-  async loadEvent(conversationId, eventId) {
+  async loadEvent(conversationId: string, eventId: string): Promise<DBEventObject> {
     if (!conversationId || !eventId) {
       this.logger.error(`Cannot get event '${eventId}' in conversation '${conversationId}' without IDs`);
-      throw new ConversationError(BaseError.TYPE.MISSING_PARAMETER, BaseError.MESSAGE.MISSING_PARAMETER);
+      throw new ConversationError(
+        BaseError.TYPE.MISSING_PARAMETER as BASE_ERROR_TYPE,
```
Comment (ffflorian) [inline src/script/event/EventService.ts]:
```suggestion
        BASE_ERROR_TYPE.MISSING_PARAMETER,
```

File: src/script/event/EventService.ts [hunk 3]
```diff
@@ -175,11 +200,11 @@ export class EventService {
    * @returns {Promise} Resolves with the retrieved records
```
Comment (ffflorian) [inline src/script/event/EventService.ts]: Please remove all type declarations from JSDoc.

File: src/script/event/EventService.ts [hunk 4]
```diff
@@ -293,78 +330,71 @@ export class EventService {
...
+    const assetData = event.data;
+    record.data.id = assetData.id;
+    record.data.key = assetData.key;
+    record.data.otr_key = assetData.otr_key;
+    record.data.sha256 = assetData.sha256;
+    record.data.status = AssetTransferState.UPLOADED;
+    record.data.token = assetData.token;
+    record.status = StatusType.SENT;
+    await this.replaceEvent(record);
+    return this.logger.info('Updated asset message_et (uploaded)', primaryKey);
```
Comment (ffflorian) [inline src/script/event/EventService.ts]:
```suggestion
    this.logger.info('Updated asset message_et (uploaded)', primaryKey);
    return;
```

File: src/script/event/EventService.ts [hunk 5]
```diff
@@ -293,78 +330,71 @@ export class EventService {
...
-      record.data.status = AssetTransferState.UPLOADED;
-      record.data.token = assetData.token;
-      record.status = StatusType.SENT;
-
-      return this.replaceEvent(record).then(() => this.logger.info('Updated asset message_et (uploaded)', primaryKey));
-    });
+  async updateEventAsUploadSucceeded(primaryKey: string, event: Object): Promise<void> {
+    const record = await this.storageService.load(StorageSchemata.OBJECT_STORE.EVENTS, primaryKey);
+    if (!record) {
+      return this.logger.warn('Did not find message to update asset (uploaded)', primaryKey);
```
Comment (ffflorian) [inline src/script/event/EventService.ts]:
```suggestion
      this.logger.warn('Did not find message to update asset (uploaded)', primaryKey);
      return;
```

File: src/script/event/EventService.ts [hunk 6]
```diff
@@ -389,11 +419,10 @@ export class EventService {
...
    * Update an event in the database and checks that the update is sequential.
    *
-   * @param {number|string} primaryKey Event primary key
-   * @param {Object} [changes={}] Changes to update message with
-   * @returns {Promise<Event>} Resolves when the message was updated in database
+   * @param primaryKey Event primary key
+   * @param changes Changes to update message with
    */
-  async updateEventSequentially(primaryKey, changes = {}) {
+  async updateEventSequentially(primaryKey: string, changes = {}): Promise<number> {
```
Comment (ffflorian) [inline src/script/event/EventService.ts]:
```suggestion
  async updateEventSequentially(primaryKey: string, changes: Object = {}): Promise<number> {
```

File: src/script/event/EventService.ts [hunk 7]
```diff
@@ -402,7 +431,8 @@ export class EventService {
 
       if (this.storageService.db) {
         // Create a DB transaction to avoid concurrent sequential update.
-        return this.storageService.db.transaction('rw', StorageSchemata.OBJECT_STORE.EVENTS, () => {
+        // TODO: the second parameter of the transaction seems wrong, exie actually wants something of type `Dexie.Table` here
```
Comment (ffflorian) [inline src/script/event/EventService.ts]:
```suggestion
        // TODO: the second parameter of the transaction seems wrong, Dexie actually wants something of type `Dexie.Table` here
```

**Review by ffflorian — CHANGES_REQUESTED** [review]

**Review by ffflorian — APPROVED** [review]

File: src/script/conversation/ConversationRepository.ts [hunk 8]
```diff
@@ -618,24 +616,26 @@ export class ConversationRepository {
...
+      upperBound,
+      Config.getConfig().MESSAGES_FETCH_LIMIT,
+    )) as EventRecord[];
+    const mappedMessageEntities = await this._addPrecedingEventsToConversation(events, conversationEntity);
+    conversationEntity.is_pending(false);
+    return mappedMessageEntities;
   }
 
-  private _addPrecedingEventsToConversation(events: any[], conversationEntity: Conversation) {
+  private _addPrecedingEventsToConversation(events: EventRecord[], conversationEntity: Conversation) {
```
Comment (lipis) [inline src/script/conversation/ConversationRepository.ts:638]: I think we tend to remove the leading underscore for private ones..

**Review by lipis — APPROVED** [review]

**Pull request merged**